### ISSUE TITLE
[Bug] "fs" node module warnings when running deck.gl with vite

### ISSUE THREAD

**Issue by danloiterton** [visgl/deck.gl#8077]
### Description

When adding deck.gl to my React / vite project, I immediately start seeing warnings spam the console. So far I have done nothing other that instantiate a new layer (I have not yet tried adding it to a map). I get these warnings with both the `MVTLayer` and `GeoJsonLayer`.

The vite troubleshooting link explains:

> When you use a Node.js module in the browser, Vite will output the following warning.
> 
> Module "fs" has been externalized for browser compatibility. Cannot access "fs.readFile" in client code.
> 
> This is because Vite does not automatically polyfill Node.js modules.
> 
> We recommend avoiding Node.js modules for browser code to reduce the bundle size, although you can add polyfills manually. If the module is imported from a third-party library (that's meant to be used in the browser), it's advised to report the issue to the respective library.

### Flavors

- [ ] Script tag
- [X] React
- [ ] Python/Jupyter notebook
- [ ] MapboxOverlay
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] ArcGIS

### Expected Behavior

It would be good if these warnings went away. I am not sure if they will cause issues down the line.

### Steps to Reproduce

This is the only code I have added so far:

```
import { MVTLayer } from '@deck.gl/geo-layers/typed';

const layer = new MVTLayer({
   data: `${TILESERVICE_URL}/{z}/{y}/{x}.pbf?key=${API_KEY}`,
   minZoom: 0,
   maxZoom: 24,
});
```

The same also happened when I tried instantiating a `GeoJsonLayer`

### Environment

deck.gl@8.9.25
vite@4.3.9
Chrome 115.0.5790.114
macOS 12.6.7

### Logs

```
Module "fs" has been externalized for browser compatibility. Cannot access "fs.readdir" in client code. See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
get @ browser-external:fs:9
(anonymous) @ fs.ts:40
browser-external:fs:9 Module "fs" has been externalized for browser compatibility. Cannot access "fs.stat" in client code. See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
get @ browser-external:fs:9
(anonymous) @ fs.ts:42
browser-external:fs:9 Module "fs" has been externalized for browser compatibility. Cannot access "fs.readFile" in client code. See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
get @ browser-external:fs:9
(anonymous) @ fs.ts:45
browser-external:fs:9 Module "fs" has been externalized for browser compatibility. Cannot access "fs.readFileSync" in client code. See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
get @ browser-external:fs:9
(anonymous) @ fs.ts:47
browser-external:fs:9 Module "fs" has been externalized for browser compatibility. Cannot access "fs.writeFile" in client code. See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
get @ browser-external:fs:9
(anonymous) @ fs.ts:49
browser-external:fs:9 Module "fs" has been externalized for browser compatibility. Cannot access "fs.writeFileSync" in client code. See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
get @ browser-external:fs:9
(anonymous) @ fs.ts:51
browser-external:fs:9 Module "fs" has been externalized for browser compatibility. Cannot access "fs.open" in client code. See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
get @ browser-external:fs:9
(anonymous) @ fs.ts:56
browser-external:fs:9 Module "fs" has been externalized for browser compatibility. Cannot access "fs.read" in client code. See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
get @ browser-external:fs:9
(anonymous) @ fs.ts:61
browser-external:fs:9 Module "fs" has been externalized for browser compatibility. Cannot access "fs.fstat" in client code. See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
get @ browser-external:fs:9
(anonymous) @ fs.ts:63
browser-external:fs:9 Module "fs" has been externalized for browser compatibility. Cannot access "fs.createWriteStream" in client code. See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
get @ browser-external:fs:9
(anonymous) @ fs.ts:65
browser-external:stream:9 Module "stream" has been externalized for browser compatibility. Cannot access "stream.Transform" in client code. See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
```

**Comment by ibgreen** [comment]
As has been explained in other discussions, the current understanding of the problem is that vite uses esbuild and esbuild doesn't respect the `browser` field in nested package dependencies, specifically in loaders.gl where it is used to exclude node specific code in browser builds. 

Would like to fix this of course, it is just a matter of competing priorities and a simple solution not being immediately obvious.

FWIW, by installing the polyfills you get rid of the warnings which is nice but I fear you may bundle node polyfill code that is never actually used. Perhaps it is tree shaken so that it evens out, but maybe check the size of your bundle with and without these polyfills before celebrating?

**Comment by Pessimistress** [comment]
Fixed in v8.9.27.
